tools/block-generator/generator/generate.go

```diff
@@ -93,8 +120,19 @@ func MakeGenerator(dbround uint64, bkGenesis bookkeeping.Genesis, config Generat
 		gen.genesisHash = bkGenesis.Hash()
 	}
 
-	gen.apps = make(map[appKind][]*appData)
-	gen.pendingApps = make(map[appKind][]*appData)
+	gen.resetPendingApps()
```

Comment:
`appSlice` is used for easy look up using a random index. `appMap` is used for $O(1)$ lookup to tell if an app exists and get it's information. Otherwise it would be $O(N)$ where $N$ is the number of created apps.

Reply:
`resetPendingApps()` also sets `gen.PendingApps[of each kind]` and `gen.PendingAppMap[of each kind]` as we  appending to totally empty maps of other types panics (as opposed to appending to a nil slice)

Reply:
@tzaffi could we move some methods to helpers file or a file that's just for boxes utils? this file is getting large and it's harder to review.

Reply:
The future follow up PR will introduce the following files to break up the code a bit:
* [generate_apps.go](https://github.com/tzaffi/go-algorand/blob/6374eb4790ece4d2bf685cab645c197f221aef17/tools/block-generator/generator/generate_apps.go)
* [generate_ledger.go](https://github.com/tzaffi/go-algorand/blob/6374eb4790ece4d2bf685cab645c197f221aef17/tools/block-generator/generator/generator_ledger.go)

I've also introduced TODO's for this in the PR description